tf-psa-crypto/tests/suites/test_suite_gcm.function

```diff
@@ -605,6 +605,146 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void gcm_encrypt_input_output_buffer_overlap(int cipher_id, data_t *key_str,
+                         data_t *src_str, data_t *iv_str,
+                         data_t *add_str, data_t *dst,
+                         int tag_len_bits, data_t *tag,
+                         int init_result)
+{
+    unsigned char *buffer;
+    size_t buffer_len;
+    unsigned char tag_output[16];
+    mbedtls_gcm_context ctx;
+    size_t tag_len = tag_len_bits / 8;
+    size_t n1;
+    size_t n1_add;
+
+    BLOCK_CIPHER_PSA_INIT();
+    mbedtls_gcm_init(&ctx);
+
+    /* GCM includes padding and therefore input length can be shorter than the output length
+     * Therefore we must ensure we round up to the nearest 128-bits/16-bytes.
+     */
+    buffer_len = src_str->len;
+    if (buffer_len % 16 != 0) {
+      buffer_len += (16 - (buffer_len % 16));
+    }
+    buffer = malloc(buffer_len);
+    if (buffer == NULL) {
+        TEST_FAIL("Could not allocate memory for input output buffer overlap");
+        goto exit;
+    }
+    memcpy(buffer, src_str->x, src_str->len);
+
+    memset(tag_output, 0x00, 16);
+
+    TEST_ASSERT(mbedtls_gcm_setkey(&ctx, cipher_id, key_str->x, key_str->len * 8) == init_result);
+    if (init_result == 0) {
+        TEST_ASSERT(mbedtls_gcm_crypt_and_tag(&ctx, MBEDTLS_GCM_ENCRYPT, src_str->len, iv_str->x,
+                                              iv_str->len, add_str->x, add_str->len, src_str->x,
```

Comment:
You're still passing non-overlapping buffers here, as `src_str->x` is a different buffer to `buffer`. I believe this should be:
```suggestion
                                              iv_str->len, add_str->x, add_str->len, buffer,
```

Reply:
Thanks, fixed now.